src/autocomplete/WildcardMentionItem.js

```diff
@@ -0,0 +1,166 @@
+import type { Node } from 'react';
+import { View } from 'react-native';
+import * as typeahead from '@zulip/shared/js/typeahead';
+
+import type { GetText, Narrow } from '../types';
+import { IconWildcardMention } from '../common/Icons';
+import ZulipText from '../common/ZulipText';
+import Touchable from '../common/Touchable';
+import { createStyleSheet, ThemeContext } from '../styles';
+import { caseNarrowDefault } from '../utils/narrow';
+import { TranslationContext } from '../boot/TranslationProvider';
+
+/**
+ * A type of wildcard mention recognized by the server.
+ *
+ * For the string the server knows this by, see serverCanonicalStringOf.
+ *
+ * See user doc on wildcard mentions:
+ *   https://zulip.com/help/pm-mention-alert-notifications#wildcard-mentions
+ */
+export enum WildcardMentionType {
+  // These values are arbitrary. For the string the server knows these by,
+  // see serverCanonicalStringOf.
+  All = 0,
+  Everyone = 1,
+}
+
+/**
+ * The canonical English string, like "everyone" or "all".
+ *
+ * See also serverCanonicalStringOf for the string the server knows this by.
+ * (It might differ in a future where servers localize these.)
+ */
+// All of these should appear in messages_en.json, so we can make the
+// wildcard mentions discoverable in the people autocomplete in the client's
+// own language. See getFilteredSuggestions.
+const englishCanonicalStringOf = (type: WildcardMentionType): string => {
+  switch (type) {
+    case WildcardMentionType.All:
+      return 'all';
+    case WildcardMentionType.Everyone:
+      return 'everyone';
+  }
+};
+
+/**
+ * The string recognized by the server, like "everyone" or "all".
+ *
+ * Currently the same as englishCanonicalStringOf, but that should change if
+ * servers start localizing these.
+ */
+export const serverCanonicalStringOf = englishCanonicalStringOf;
+
+export const descriptionOf = (
+  type: WildcardMentionType,
+  destinationNarrow: Narrow,
+  _: GetText,
+): string => {
+  switch (type) {
+    case WildcardMentionType.All:
+    case WildcardMentionType.Everyone:
+      return caseNarrowDefault(
+        destinationNarrow,
+        { topic: () => _('Notify stream'), pm: () => _('Notify recipients') },
+        // A "destination narrow" should really be a conversation narrow
+        // (i.e., stream or topic), but including this default case is easy,
+        // and better than an error/crash in case we've missed that somehow.
+        () => _('Notify everyone'),
+      );
+  }
+};
+
+export const getFilteredSuggestions = (
+  query: string,
+  destinationNarrow: Narrow,
+  _: GetText,
+): $ReadOnlyArray<WildcardMentionType> =>
+  Array.from(WildcardMentionType.members()).filter(type =>
+    /* $FlowFixMe[prop-missing] - Type query_matches_source_attrs. In
+       particular, find a way to require `source` to have string values at
+       the keys given in `match_attrs`. */
+    typeahead.query_matches_source_attrs(
```

Comment:
Thanks! I've published that change to NPM and my next revision uses `typeahead.query_matches_string` instead of `typeahead.query_matches_source_attrs`.